### DISCUSSION TITLE
how to iterate over multi map values in a process?

### DISCUSSION THREAD

**Discussion by complexgenome** [nextflow-io/nextflow#4650] [Q&A] [answered]
I've following code.

```
process test {
    input:
    val (it)
    output:
    stdout
    script:
    println(it)
    """
    echo ${it}
    """

}
workflow {

Channel.of(1, 2, 3, 4)
    .multiMap { it ->
        foo: it + 1
        bar: it * it
    }
    .set { result }

//result.foo.view { "foo $it" }
//result.bar.view { "bar $it" }
test(result.foo).view()
}

```
When I run it:
nextflow run script.nf

It prints `null` string. How do I enable it to print correct values?

**Comment by robsyme** [comment, marked as answer]
The issue here is that the `it` variable is reserved for the iterator value and should not be used in the process scope.

This version of your workflow should produce the expected result:


```groovy
process test {
    input: val(num)
    output: stdout
    script: "echo ${num}"
}

workflow {
    Channel.of(1, 2, 3, 4)
    .multiMap {
        foo: it + 1
        bar: it * it
    }
    .set { result }

    test(result.foo).view()
}
```

In future, if your query is a question related to Nextflow syntax, please consider the Nextflow community forum: https://community.seqera.io.

If you query is a Nextflow bug report, this Github repository is still the appropriate venue.

**Comment by complexgenome** [comment]
@robsyme Thank you for your reply. I will use the shared link for my errors/syntax related stuff. I didn't know of the platform. I've been waiting for help on many posts on this github Q&A disucssion forum.
